agent/envoyextensions/builtin/ext-authz/structs.go

```diff
@@ -652,7 +652,10 @@ func (t *Target) validate() error {
 		addr := strings.Split(t.URI, ":")
 		if len(addr) == 2 {
 			t.host = addr[0]
-			if t.host != "localhost" && t.host != "127.0.0.1" {
+			if t.host == "localhost" || t.host == "127.0.0.1" {
```

Comment:
`localhost` can resolve to an IPv6 address (i.e., `::1`) if the underlying host is v6-enabled.

If Consul is always going to explicitly translate `localhost` to the v4 loopback address, we should explicitly call out that behavior in the docs.

Reply:
Could we not instead detect that a hostname is used and use either a STRICT_DNS or LOGICAL_DNS cluster instead?

Generally we should try to be as IP protocol agnostic as possible. While many of our existing APIs make that difficult, our future resource model is much more flexible.

Besides ext_authz I could see needing to configure other aspects of Envoy using hostnames (SDS providers, location where to download WASM plugins, JWKS urls) and having better support for dual stack DNS clusters for hostname based resolution seems like it is the way to go.

If that would be too great a lift for right now, we should track this as some follow on work to shore up during the next release cycle.

Reply:
I'm all for setting the cluster type dynamically if we detect the `localhost` hostname instead of an IP address. It is a pretty easy lift and seems like a much more flexible solution in the long term.